### PR TITLE
VIH-9293 - Add custom mount path for key vaults

### PR DESCRIPTION
**Pull request by chris-pateman**
### JIRA link (if applicable) ###
VIH-9293

### Change description ###
Add custom mount path for the key vaults.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: library/Chart.yaml [hunk 1]
```diff
@@ -2,17 +2,26 @@ apiVersion: v2
...
 # a dependency of application charts to inject those utilities and functions into the rendering
+
 # pipeline. Library charts do not define any templates and therefore cannot be deployed.
 type: library
 # This is the chart version. This doesn't need to be incremented on every change. 
+
 # It is managed by release pipeline based on Github tag
 version: 0.1.0 # Do not change
 # This is the version number of the application being deployed. This doesn't need to be incremented on every change.
+
```
Comment (timja) [inline library/Chart.yaml:25]: revert all these extra blank lines generally it's making it harder to read

File: library/templates/v2/_deployment_test_helpers.tpl [hunk 2]
```diff
@@ -89,10 +89,10 @@ containers:
     {{- end }}
     {{- end }}
     {{- if and .Values.testsConfig.keyVaults .Values.global.enableKeyVaults }}
-    volumeMounts:
+    volumeMounts:
```
Comment (timja) [inline library/templates/v2/_deployment_test_helpers.tpl:92]:
```suggestion
    volumeMounts:
```

File: library/templates/v2/_secretcsivolumes.tpl [hunk 3]
```diff
@@ -25,7 +25,7 @@ volumes:
 {{- end }}
 
 {{/*
-Mount the Key vaults on /mnt/secrets
+Mount the Key vaults on /mnt/secrets by Default or the custom path
```
Comment (timja) [inline library/templates/v2/_secretcsivolumes.tpl:28]:
```suggestion
Mount the Key vaults on /mnt/secrets by default or the custom mountPath
```

File: tests/results/secretproviderclass.yaml [hunk 4]
```diff
@@ -34,3 +34,21 @@ spec:
...
+  parameters:
+    userAssignedIdentityID: ""
+    usePodIdentity: "true"
+    keyvaultName: "bulk-scan-v2-aat"
+    objects: |
+      array:
+        - |
+          objectName: alias-secret
+          objectType: secret
+    tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
```
Comment (timja) [inline tests/results/secretproviderclass.yaml:54]:
```suggestion
    tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"

```

**Review by timja — CHANGES_REQUESTED** [review]
I'm not 100% sure we want this, as still our recommendation is to adapt the application.

The current minor style issues with the PR are below

File: tests/results/secretproviderclass.yaml (hunk 4 above)
Comment (chris-pateman) [reply tests/results/secretproviderclass.yaml]: Corrected

File: library/templates/v2/_secretcsivolumes.tpl (hunk 3 above)
Comment (chris-pateman) [reply library/templates/v2/_secretcsivolumes.tpl]: Corrected

File: library/templates/v2/_deployment_test_helpers.tpl (hunk 2 above)
Comment (chris-pateman) [reply library/templates/v2/_deployment_test_helpers.tpl]: Corrected

File: library/Chart.yaml (hunk 1 above)
Comment (chris-pateman) [reply library/Chart.yaml]: Corrected

**Review by timja — APPROVED** [review]

**Pull request merged**